src/fluent/watch.ts

```diff
@@ -220,11 +220,126 @@ export class Watcher<T extends GenericClass> {
     return { opts, url };
   };
 
-  /** Run the watch. */
-  #runner = async () => {
+  /**
+   * Retrieve the list of resources and process the events.
+   *
+   * @param continueToken - the continue token for the list
+   * @param removedItems - the list of items that have been removed
+   */
+  #list = async (continueToken?: string, removedItems?: Map<string, InstanceType<T>>) => {
+    try {
+      const { opts, url } = await this.#buildURL(false, undefined, continueToken);
+
+      // Make the request to list the resources
+      const response = await wrappedFetch<KubernetesListObject<InstanceType<T>>>(url, opts);
+      const list = response.data;
+
+      // If the request fails, emit an error event and return
+      if (!response.ok) {
+        this.#events.emit(
+          WatchEvent.LIST_ERROR,
+          new Error(`list failed: ${response.status} ${response.statusText}`),
+        );
+
+        return;
+      }
+
+      // Gross hack, thanks upstream library :<
+      if ((list.metadata as { continue?: string }).continue) {
+        continueToken = (list.metadata as { continue?: string }).continue;
+      }
+
+      // Emit the list event
+      this.#events.emit(WatchEvent.LIST, list);
+
+      // Update the resource version from the list metadata
+      this.#resourceVersion = list.metadata?.resourceVersion;
+
+      // If removed items are not provided, clone the cache
+      removedItems = removedItems || new Map(this.#cache.entries());
+
+      // Process each item in the list
+      for (const item of list.items || []) {
+        const { uid } = item.metadata;
+
+        // Remove the item from the removed items list
+        const alreadyExists = removedItems.delete(uid);
+
+        // If the item does not exist, it is new and should be added
+        if (!alreadyExists) {
+          // Send aded event. Use void here because we don't care about the result (no consequences here if it fails)
+          void this.#process(item, WatchPhase.Added);
+          continue;
+        }
+
+        // Check if the resource version has changed for items that already exist
+        const cachedRV = parseInt(this.#cache.get(uid)?.metadata?.resourceVersion);
+        const itemRV = parseInt(item.metadata.resourceVersion);
+
+        // Check if the resource version is newer than the cached version
+        if (itemRV > cachedRV) {
```

Comment:
This check doesn't really say if there has been _more than one modification_ in the time between the cachedRV was set and it itemRV was iterated though... right?

If that's the case, then line 282 is going to fire a single MODIFIED event for _one-or-more_ intermediate changes.  Is that the guarantee we're trying to make?  Or maybe I'm misreading here?